include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -82,6 +82,170 @@ namespace internal
     {}
 
 
+
+    namespace
+    {
+      template <int dim, int spacedim>
+      unsigned int
+      setup_parallel_data_base_internal(
```

Comment:
this function could use a comment about what it does and what it returns (directly or through its non-`const` reference arguments).